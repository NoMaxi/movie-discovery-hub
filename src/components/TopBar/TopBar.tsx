import { NetflixRouletteText } from "@/components/common/NetflixRouletteText/NetflixRouletteText";

interface TopBarProps {
    onAddMovieClick: () => void;
}

export const TopBar = ({ onAddMovieClick }: TopBarProps) => (
    <div
        className="
            flex justify-between items-center fixed
            top-0 left-0 right-0 h-3xl px-16 z-50
            bg-opacity-80 backdrop-blur-sm shadow-md
        "
    >
        <NetflixRouletteText />
        <button className="btn-add-movie" onClick={onAddMovieClick}>
            + Add Movie
        </button>
    </div>
);
