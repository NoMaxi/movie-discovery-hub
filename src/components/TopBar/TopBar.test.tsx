import { render, screen } from "@testing-library/react";
import userEvent, { UserEvent } from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { TopBar } from "./TopBar";

jest.mock("@/components/common/NetflixRouletteText/NetflixRouletteText", () => ({
    NetflixRouletteText: jest.fn(() => <div data-testid="mocked-netflix-roulette-text">NetflixRouletteText</div>),
}));

describe("TopBar", () => {
    let onAddMovieClickMock: jest.Mock;
    let user: UserEvent;

    beforeEach(() => {
        onAddMovieClickMock = jest.fn();
        user = userEvent.setup();
    });

    test("Should render correctly with NetflixRouletteText and Add movie button", () => {
        const { asFragment } = render(<TopBar onAddMovieClick={onAddMovieClickMock} />);
        const netflixRouletteText = screen.getByTestId("mocked-netflix-roulette-text");

        expect(netflixRouletteText).toBeInTheDocument();

        const addButton = screen.getByRole("button", { name: /\+ add movie/i });

        expect(addButton).toHaveClass("btn-add-movie");
        expect(asFragment()).toMatchSnapshot();
    });

    test("Should call 'onAddMovieClick' prop when the Add movie button is clicked", async () => {
        render(<TopBar onAddMovieClick={onAddMovieClickMock} />);
        const addButton = screen.getByRole("button", { name: /\+ add movie/i });

        await user.click(addButton);

        expect(onAddMovieClickMock).toHaveBeenCalledTimes(1);
    });
});
