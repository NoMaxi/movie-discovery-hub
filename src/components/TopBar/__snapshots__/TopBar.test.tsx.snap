// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Should render correctly with NetflixRouletteText and Add movie button 1`] = `
<DocumentFragment>
  <div
    class="
            flex justify-between items-center fixed
            top-0 left-0 right-0 h-[60px] px-16 z-50
            bg-opacity-80 backdrop-blur-sm shadow-md
        "
  >
    <div
      data-testid="mocked-netflix-roulette-text"
    >
      NetflixRouletteText
    </div>
    <button
      class="btn-add-movie"
    >
      + Add Movie
    </button>
  </div>
</DocumentFragment>
`;
